jiant/allennlp_mods/span_metrics.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+import torch
+
+from allennlp.training.metrics.metric import Metric
+
+
+@Metric.register("spanf1")
+class SpanF1Measure(Metric):
```

Comment:
any reason not to use the allennlp version? 

https://github.com/allenai/allennlp/blob/master/allennlp/training/metrics/span_based_f1_measure.py

Reply:
It looks like AllenNLP has two options. The one you linked which is based on Conll SRL, which is based on BIO (+variants) style of labels, and https://github.com/allenai/allennlp/blob/master/allennlp/training/metrics/squad_em_and_f1.py for SQuAD, but is based on raw strings. Neither is quite what we are looking for (token-based EM/F1 for spans). Arguably we can reformulate the labels to match the BIO variant (I'm not too familiar with this), but let me know what you think.